### PR TITLE
ci: add support label to issues with support reference

### LINKED ISSUE #12655: Automatically add `support` label to support related issues
**Is your feature request related to a problem? Please describe.**
After a release we must send a list of fixed issues related to support tickets. For this we look at the `support` label on issues. It's easy to forget to add this label.

**Describe the solution you'd like**
Introduce a GitHub action that checks new issues and comments in issues if the text contains `SUPPORT-XXXX`. If it find any the action should add the `support` label.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by megglos**
## Description
Will remove automate adding a `support` label to issues and just require us to mention/link the Jira issue id which needs to be done anyway according to https://confluence.camunda.com/display/HAN/C8+Support+to+Product+Escalation+Process#C8SupporttoProductEscalationProcess-ProductEngineer-Process

## Related issues

closes #12655

### CODE REVIEW

File: .github/issue-labeler.yml
Comment (megglos) [inline .github/issue-labeler.yml:3]: should trigger on just id's mentioned but also the links like `https://jira.camunda.com/browse/SUPPORT-99999`

File: .github/workflows/issue-add-support-label.yml
Comment (remcowesterhoud) [inline .github/workflows/issue-add-support-label.yml:7]: ❌ We should also check for `issue_comments` when `created` or `edited`. We sometimes have a known issue before a support ticket is created. Usually a comment will be placed in the issue linking to the support case.

**Review by remcowesterhoud — APPROVED** [review]
Thanks @megglos! I feel we need 1 more small change. Feel free to merge it after without second review.

File: .github/workflows/issue-add-support-label.yml
Comment (megglos) [reply .github/workflows/issue-add-support-label.yml:7]: I seems right now this is not supported by the issue labeler action https://github.com/github/issue-labeler/issues/30 :/ 

Generally I would be in favor of always linking the support issue in the issue content, rather than in some comment, to keep all the meta-information on an issue in the main content rather than spread it across comments.
I would adjust the linking instructions on https://confluence.camunda.com/display/HAN/C8+Support+to+Product+Escalation+Process#C8SupporttoProductEscalationProcess-ProductEngineer-CrosslinkJiraandGithubissues accordingly. wdyt?

Comment (megglos) [reply .github/workflows/issue-add-support-label.yml:7]: ok, I adjusted the instructions and will also announce/highlight this in the support process slack channel.
If we later on decide this isn't practical we could expand on scanning comments.

Comment (remcowesterhoud) [reply .github/workflows/issue-add-support-label.yml:7]: Sounds good to me 👍

**Comment by megglos** [comment]
bors merge

**Comment by megglos** [comment]
bors retry

**Pull request merged**
